libraries/AP_HAL/utility/RingBuffer.h

```diff
@@ -167,7 +168,7 @@ class ObjectBuffer {
      * push_force() is semantically equivalent to:
      *   if (!push(t)) { pop(); push(t); }
      */
-    bool push_force(const T &object) {
+    virtual bool push_force(const T &object) {
```

Comment:
@tridge has pointed out that this code turns a writer into a reader, which is unsafe.

Reply:
so we should just have a new class rather than overriding? I guess I am overriding every function in any-case so its some what pointless to inherit.

Reply:
@IamPete1 This is a preexsisting problem he's highlighting, it's the core problem that prompted #13217. The main comment here is actually that `push_force` will only be safe if you are using it on the locking version you just made. So in this case it's actually probably better to remove push_force here, and move it's entire implementation into your new locking variant.